ProjFS.Mac/PrjFSKext/PrjFSKext/KauthHandler.cpp

```diff
@@ -376,19 +379,66 @@ static int HandleFileOpOperation(
     atomic_fetch_add(&s_numActiveKauthEvents, 1);
     
     vfs_context_t context = vfs_context_create(NULL);
+    vnode_t currentVnodeFromPath = NULLVP;
 
-    if (KAUTH_FILEOP_CLOSE == action)
+    if (KAUTH_FILEOP_RENAME == action ||
+        KAUTH_FILEOP_LINK == action)
     {
-        vnode_t currentVnode = reinterpret_cast<vnode_t>(arg0);
-        // arg1 is the (const char *) path
-        int closeFlags = static_cast<int>(arg2);
+        // arg0 is the (const char *) fromPath (or the file being linked to)
+        const char* newPath = (const char*)arg1;
         
-        vtype vnodeType = vnode_vtype(currentVnode);
-        if (ShouldIgnoreVnodeType(vnodeType, currentVnode))
+        // TODO(Mac): Improve error handling
```

Comment:
And more importantly, we just lost a notification here. We will need some pattern for dealing with that.

Reply:
Good point, I will add some more details to this comment